packages/@blockprotocol/graph/src/stdlib.ts

```diff
@@ -109,12 +109,36 @@ type GenericLinkEntityAndRightEntity = {
  * @param {EntityId} entityId - The ID of the source entity to search for outgoing links from
  */
 export const getOutgoingLinkAndTargetEntities = <
-  LinkAndRightEntities extends GenericLinkEntityAndRightEntity[] = GenericLinkEntityAndRightEntity[],
+  LinkAndRightEntities extends LinkEntityAndRightEntity[] = LinkEntityAndRightEntity[],
 >(
   subgraph: Subgraph,
   entityId: EntityId,
 ): LinkAndRightEntities =>
-  getOutgoingLinkAndTargetEntitiesTemporal(
+  getOutgoingLinkAndTargetEntitiesTemporal<LinkAndRightEntities>(
     subgraph,
     entityId,
-  ) as LinkAndRightEntities; // @todo add generics elsewhere
+  );
+
+/**
+ * Adds simple property access to an entity, MUTATING the object passed in.
+ * The simple accessors are the last path segment before the trailing slash of a base URI
+ * e.g. if there is a key "https://example.com/my-type/" then a "my-type" key is added
+ * @todo is this helper actually helpful, or should people just deal with accessing by URI?
+ *    - they'll still have to deal with URIs, e.g. when sending updates
+ */
+export const addSimpleAccessors = <
+  Properties extends EntityPropertiesObject = EntityPropertiesObject,
+>(
+  entity: Entity<Properties>,
+) => {
+  for (const [key, value] of Object.entries(entity.properties)) {
+    const [simpleAccessor, trailingSpace] = key.split("/").slice(-2);
+    if (trailingSpace !== "" || !simpleAccessor) {
+      throw new Error(`Property key ${key} is not a valid base URI.`);
+    }
+    // eslint-disable-next-line no-param-reassign -- intentional mutation
+    entity.properties[simpleAccessor] = value;
```

Comment:
If we do want a function such as this, this is the wrong way to do it, I've belatedly realise – it will have surprising results if people are spreading out the object.

The better way would be to create a `Proxy` with a `get` trap. I didn't do that first time round because it would have worse performance, because you'd need to check the key on every access rather than just throwing the new keys on at the start. 

I'm not going to update this yet until we've discussed if this is worth having at all.

Reply:
I think we definitely need simple accessors, but as I mentioned in passing a while ago, we should _**not**_ be relying on the URL format. If we generate simple accessors they should be based on the `Title` property of the schemas.

I agree about your `Proxy` approach, and I also do not think we should be putting this method here alongside the `Subgraph` library functions. (Not suggesting we have to limit the `stdlib` to the `Subgraph` type but we should restructure the package otherwise.)

Reply:
I think for now we should probably remove this from the PR, log it as a potential approach, and create a follow-up to investigate the correct solution.

Reply:
I've removed the function and renamed the type in https://github.com/blockprotocol/blockprotocol/pull/866/commits/f1d4077f10d4c09a41baf5455f030d81d8d2026d– I'd like to keep the type around in case to experiment with it in block buiding to see if I can figure out some better pattern. We can remove it if it turns out to be not helpful